### PR TITLE
cap urllib3 to avoid openssl issue

### PR DESCRIPTION
**Pull request by jburel**
Problem noticed when running the omero-server job on CentOS 7 with SCL Python 3.8
cc @sbesson @khaledk2

### CODE REVIEW

**Comment by sbesson** [comment]
Cross-linking to https://github.com/ome/scc/pull/284 which patched a similar issue (in a build-only dependency) and https://urllib3.readthedocs.io/en/stable/v2-migration-guide.html which explains the issues associated with `urllib3 2.x`.

This problem will primarily impact CentOS 7 deployments with SCL Python 3.8 and OpenSSL 1.0.x. Unfortunately I am not aware of a classifier that would detect this environment and fallback to `urllib3 1.x`. 

The primary downside of this change is that the capping `urrlib3` will happen across the board including in more recent distributions where OpenSSL 1.1.1 or even 3.x are shipped with Python 3.7+. Looking at other changes, it looks like `urllib3 2.x` also deprecated the `urllib3.contrib.pyopenssl` module which was used as the basis of https://github.com/ome/omero-py/pull/305.

In summary:

- this is yet another incentive to decide and communicate on the CentO7 support timeline - see https://github.com/ome/omero-certificates/pull/33 for another SSL issue related to this environment
- more investigation is required on the urllib3 2.x breaking changes and their impact on the OMERO Python stack
